safe_transaction_service/history/serializers.py

```diff
@@ -266,8 +270,14 @@ def validate(self, attrs):
         signature_owners = []
         # TODO Make signature mandatory
         signature = attrs.get("signature", b"")
+        # For v1.5.0+, the isValidSignature(bytes32,bytes) expects the original message_hash (bytes32),
+        safe_signature_hash = (
```

Comment:
I'm missing something here. You defined this function `select_preimage_by_safe_version`, that I agree is required, to select the preimage for signatures. v1.4.1 used `preimage` and v1.5.0 uses `keccak(preimage)`  But I don't understand why here https://github.com/safe-global/safe-transaction-service/pull/2714/files#diff-ffa3f7602acc3a07a0d3bb820224a2d4a1c49e4c0a40bf8d3f32e8d3111cbbb5R127 you used the function and now you do an `if`

Reply:
replaced with `select_safe_encoded_message_hash_by_safe_version`